api/src/main/java/org/openmrs/module/auditlogweb/api/dao/AuditDao.java

```diff
@@ -115,4 +139,253 @@ public <T> AuditEntity<T> getAuditEntityRevisionById(Class<T> entityClass, int e
+            Integer userId, Date startDate, Date endDate) {
+        return getRevisionsWithFilters(entityClass, page, size, userId, startDate, endDate, "desc");
+    }
+
+    /**
+     * Counts the number of entity revisions that match the given filters.
+     *
+     * @param entityClass the class of the audited entity
+     * @param userId      optional user ID to filter changes by user
+     * @param startDate   optional filter to include changes from this date onward
+     * @param endDate     optional filter to include changes up to this date
+     * @param <T>         the type of the audited entity
+     * @return the count of matching revisions
+     */
+    public <T> long countRevisionsWithFilters(Class<T> entityClass, Integer userId, Date startDate, Date endDate) {
+        AuditReader reader = AuditReaderFactory.get(sessionFactory.getCurrentSession());
+        AuditQuery query = EnversUtils.buildCountQueryWithFilters(reader, entityClass, userId, startDate, endDate);
+        Number countResult = (Number) query.getSingleResult();
+        return countResult != null ? countResult.longValue() : 0L;
+    }
+
+    /**
+     * Helper method to convert raw Envers result arrays into {@link AuditEntity} objects.
+     *
+     * @param entityClass the audited entity class
+     * @param auditResult an Object[] array containing entity, revision, and revision type
+     * @param <T>         the type of the audited entity
+     * @return a fully populated {@link AuditEntity}
+     */
+    private <T> AuditEntity<T> mapToAuditEntity(Class<T> entityClass, Object[] auditResult) {
+        T entity = entityClass.cast(auditResult[0]);
+        OpenmrsRevisionEntity revision = (OpenmrsRevisionEntity) auditResult[1];
+        RevisionType type = (RevisionType) auditResult[2];
+        return new AuditEntity<>(entity, revision, type, revision.getChangedBy());
+    }
+
+    /**
+     * Retrieves paginated audit entries across all dynamically discovered audited entity classes.
+     *
+     * @param page the page number (0-based)
+     * @param size number of records per page
+     * @param userId optional user ID filter
+     * @param startDate optional start date filter
+     * @param endDate optional end date filter
+     * @return paginated list of {@link AuditEntity} records
+     */
+    public List<AuditEntity<?>> getAllRevisionsAcrossEntities(int page, int size, Integer userId, Date startDate, Date endDate, String sortOrder) {
+        List<String> auditedClassNames = UtilClass.findClassesWithAnnotation()
+                .stream()
+                .filter(className -> {
+                    try {
+                        Class<?> clazz = Class.forName(className);
+                        return !Modifier.isAbstract(clazz.getModifiers());
+                    } catch (ClassNotFoundException e) {
+                        return false;
+                    }
+                })
+                .collect(Collectors.toList());
```

Comment:
It will be better to extract this into a new method. This is also getting used in "countRevisionsAcrossEntities"